doc/installingNative.md

```diff
@@ -1,78 +1,100 @@
+This page describes one way to install and run PrairieLearn without any use of Docker on MacOS.
 
-* Install the pre-requisites:
+* Install XCode Command-Line Tools to get command-line `git` (among other things):
 
-    * [Node.js](http://nodejs.org/) version  or higher
-    * [npm](https://npmjs.org/) (included with Node.js on Windows)
-    * [PostgreSQL](https://www.postgresql.org) version 9.6 or higher
-    * [Python 3](https://www.python.org) version 3.6 or higher
-    * command-line git or [GitHub Desktop](https://desktop.github.com)
+```sh
+xcode-select --install
+```
 
-On OS X these can be installed with [MacPorts](http://www.macports.org/) or [Homebrew](http://brew.sh/). On Linux these should all be standard packages from the OS distrbution.
+* Install [Node.js](http://nodejs.org/) by downloading and running the installation package. This will get you both `node` and `npm`.
 
-Note that with MacPorts you need to select the active version of PostgreSQL, for example `port select postgresql postgresql96`.
+* Install [homebrew](https://brew.sh).
 
-* Clone the latest code:
+* Install [miniconda](https://conda.io/docs/user-guide/install/macos.html#).
+
+* Use `git` to clone the PrairieLearn repo (put it wherever you want), and then `cd` to that directory:
 
 ```sh
 git clone https://github.com/PrairieLearn/PrairieLearn.git
 ```
 
-* Install the Node.js libraries:
+Use `git pull` as usual to keep your clone up to date.
+
+* Use homebrew to install [R](https://www.r-project.org):
 
 ```sh
-cd PrairieLearn
-npm install
+brew install r
+```
+
+* Use homebrew to install [PostgreSQL](https://www.postgresql.org):
+
+```sh
+brew install postgresql
 ```
 
-On OS X, it is possible that this process will fail on `node-gyp rebuild` during the install of `mcrypt`. In this case, use the following command instead:
+* Use conda to create a virtual environment (e.g., with name `PL`):
 
 ```sh
-npm install --python=PATH_TO_PYTHON2
+conda create -n PL python=3.6
+conda activate PL
 ```
 
-For example, this might be:
+* Make sure `python`, `python3`, and `python3.6` will all run the right version:
 
 ```sh
-npm install --python=/usr/bin/python2.7
+python --version      # should return "Python 3.6.x"
+python3 --version     # should return "Python 3.6.x"
+python3.6 --version   # should return "Python 3.6.x"
 ```
 
-* Make sure `python3` and `python3.6` will run the right version, and make executable links if needed:
+* Install the python packages that are used on the production server:
 
 ```sh
-python3 --version     # should return "Python 3.6" or higher
-python3.6 --version   # should return "Python 3.6" or higher
+cd PrairieLearn
+python -m pip install -r environments/centos7-plbase/python-requirements.txt
 ```
 
-* Install the Python libraries:
+* (Optionally, install the R packages that are used on the production server, as in `environments/centos7-plbase/r-requirements.R`.)
```

Comment:
(I used `Rscript ...` without any need for `chmod +x ...` - is it essential?)

When I tried this, I encountered some "permission denied" errors, like the ones discussed here:
https://github.com/Homebrew/homebrew-core/issues/17098

Thoughts?

Also, is it correct that this will install packages globally? If so, is there any alternative that will contain this install to an environment, for example a `conda` environment?

Thanks!